python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -1920,6 +1931,30 @@ def plot_events(self):
 
         self._add_page(name='event_log', html_body=body_html, title="Event Log")
 
+    def extract_times_before_reset(self):
+        # Iterate backwards over indices to extract resets and the P1 times before them.
+        curr_reset_time = None
+        get_time_before_reset = False
+
+        times_before_resets = {}
+        file_index = self.reader.get_index()
+        for entry in file_index[::-1]:
+            if entry.type == MessageType.EVENT_NOTIFICATION or get_time_before_reset:
+                # Parse entry at index for payload.
+                header, payload = self.reader.reader.parse_entry_at_index(entry)
+                system_time_ns = payload.get_system_time_ns()
+
+                if get_time_before_reset and payload.get_p1_time() is not None:
+                    times_before_resets[curr_reset_time] = payload.get_p1_time().seconds
```

Comment:
Better to use `float(payload.get_p1_time())` to get at the second value. That way, if we change the internal definition in the future (e.g., store integer and fractional parts separately), this code doesn't need to know about that.